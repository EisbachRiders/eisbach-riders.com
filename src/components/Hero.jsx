import React from 'react'
import PropTypes from 'prop-types'
import { withTranslation } from 'react-i18next'
import Img from 'gatsby-image'
import classnames from 'classnames'
import withRoot from '../withRoot'
import { withStyles } from '@material-ui/core/styles'
import Hidden from '@material-ui/core/Hidden'
import Typography from '@material-ui/core/Typography'

const styles = theme => ({
  root: {
    height: 250,
    marginTop: -65,
    [theme.breakpoints.up('sm')]: {
      height: 300,
    },
    [theme.breakpoints.up('md')]: {
      height: 700,
    },
  },
  img: {
    zIndex: -2,
    height: 250,
    [theme.breakpoints.up('sm')]: {
      height: 300,
    },
    [theme.breakpoints.up('md')]: {
      height: 700,
    },
  },
  container: {
    position: 'absolute',
    width: '100%',
    top: 65,
    paddingTop: 40,
    paddingLeft: 15,
    paddingRight: 15,
    textAlign: 'center',
    [theme.breakpoints.up('sm')]: {
      paddingTop: 130,
    },
    [theme.breakpoints.up('md')]: {
      paddingTop: 250,
    },
  },
  text: {
    textTransform: 'capitalize',
    letterSpacing: 3,
    fontWeight: 600,
    color: theme.status.white,
    textShadow: '2px 2px 4px #404040',
  },
  title: {
    fontSize: 35,
    [theme.breakpoints.up('sm')]: {
      fontSize: 48,
    },
    [theme.breakpoints.up('md')]: {
      fontSize: 96,
    },
  },
  subtitle: {
    fontSize: 14,
    [theme.breakpoints.up('sm')]: {
      fontSize: 18,
    },
    [theme.breakpoints.up('md')]: {
      fontSize: 24,
    },
  },
  logo: {
    width: 90,
    paddingBottom: 20,
    textAlign: 'center',
  },
})

const Hero = props => {
  const { img, logo, classes, t } = props

  return (
    <div className={classes.root}>
      <Img fluid={img.node.fluid} alt="" className={classes.img} />
      <div className={classes.container}>
        <Hidden smUp>
          <Img fluid={logo.node.fluid} alt="" className={classes.logo} />
        </Hidden>
        <Typography
          variant="h2"
          gutterBottom
          className={classnames(classes.text, classes.title)}
        >
          Eisbach Riders
        </Typography>
        <Typography className={classnames(classes.text, classes.subtitle)}>
          {t('hero.subtitle')}
        </Typography>
      </div>
    </div>
  )
}

Hero.propTypes = {
  classes: PropTypes.object.isRequired,
}

export default withTranslation()(withRoot(withStyles(styles)(Hero)))
