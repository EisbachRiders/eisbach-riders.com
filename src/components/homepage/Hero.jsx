import React from "react"
import { useStaticQuery, graphql } from "gatsby"
import Img from "gatsby-image"
import { useTranslation } from "react-i18next"
import clsx from "clsx"
import { makeStyles } from "@material-ui/core/styles"
import Typography from "@material-ui/core/Typography"

const useStyles = makeStyles((theme) => ({
  root: {
    position: "relative",
    height: "100vh - 98px",
  },
  container: {
    position: "absolute",
    top: "70%",
    left: "60%",
    transform: "translate(-50%, -50%)",
    textAlign: "center",
    [theme.breakpoints.up("sm")]: {
      top: "40%",
      left: "70%",
      transform: "translate(-50%, -50%)",
      width: "fit-content",
    },
    [theme.breakpoints.up("lg")]: {
      top: "50%",
      left: "70%",
      transform: "translate(-50%, -50%)",
      padding: 30,
      width: "fit-content",
      textAlign: "center",
    },
  },
  img: {
    height: "100vh",
    marginTop: -98,
  },
  text: {
    color: theme.color.black,
    fontFamily: "secondary",
  },
  textBig: {
    [theme.breakpoints.up("sm")]: {
      fontSize: 72,
    },
    [theme.breakpoints.up("lg")]: {
      fontSize: 85,
    },
    [theme.breakpoints.up("xl")]: {
      fontSize: 150,
    },
  },
  indent: {
    [theme.breakpoints.up("xs")]: {
      marginLeft: 60,
    },
    [theme.breakpoints.up("sm")]: {
      marginBottom: 60,
    },
  },
  title: {
    textTransform: "capitalize",
    fontSize: 52,
    fontWeight: 600,
    [theme.breakpoints.up("sm")]: {
      textAlign: "left",
      fontSize: 48,
    },
    [theme.breakpoints.up("lg")]: {
      fontSize: 58,
    },
    [theme.breakpoints.up("xl")]: {
      fontSize: 100,
    },
  },
  subtitle: {
    textTransform: "lowercase",
    textAlign: "center",
    fontSize: 18,
    [theme.breakpoints.up("md")]: {
      fontSize: 20,
    },
    [theme.breakpoints.up("xl")]: {
      fontSize: 34,
    },
  },
}))

function Hero() {
  const classes = useStyles()
  const { t } = useTranslation()
  const data = useStaticQuery(graphql`
    query {
      fileName: file(relativePath: { eq: "hero.jpg" }) {
        childImageSharp {
          fluid {
            ...GatsbyImageSharpFluid
          }
        }
      }
    }
  `)

  return (
    <div className={classes.root}>
      <Img
        fluid={data.fileName.childImageSharp.fluid}
        alt="wave"
        placeholderStyle={{ backgroundColor: `blue` }}
        className={classes.img}
        imgStyle={{ objectPosition: "center center" }}
      />
      <div className={classes.container}>
        <Typography
          variant="h1"
          gutterBottom
          className={clsx(classes.text, classes.title)}
        >
          <span className={classes.textBig}>E</span>isbach
        </Typography>
        <Typography
          variant="h1"
          gutterBottom
          className={clsx(classes.text, classes.title, classes.indent)}
        >
          <span className={classes.textBig}>R</span>iders
        </Typography>

        <Typography
          component="h2"
          className={clsx(classes.text, classes.subtitle)}
        >
          {t("homepage.heroSubtitle")}
        </Typography>
      </div>
    </div>
  )
}

export default Hero
