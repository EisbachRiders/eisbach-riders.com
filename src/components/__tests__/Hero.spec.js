import { render } from 'react-testing-library'
import Hero from '../Hero'

describe('Hero', () => {
  it('renders correctly', () => {
    const component = render(<Hero />)
    expect(component).toMatchSnapshot()
  })
})
