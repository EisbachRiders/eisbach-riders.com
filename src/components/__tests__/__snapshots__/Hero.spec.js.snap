// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Hook-root-xpbc82"
      >
        <div
          class="Hook-img-2zpuhd gatsby-image-wrapper"
          style="position: absolute; overflow: hidden;"
        >
          <div
            style="width: 100%;"
          />
          <picture>
            <img
              alt="wave"
              sizes="mockString"
              src="mockString"
              srcset="mockString"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; opacity: 0;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;img sizes="mockString" srcset="mockString" src="mockString" alt="wave" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
        <div
          class="Hook-container-gxkfpg"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom Hook-text-t3rl6s Hook-title-trdj2x"
          >
            E
            <span
              class="Hook-small-e5sp6t"
            >
              ISBACH
            </span>
          </h2>
          <h2
            class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom Hook-text-t3rl6s Hook-title-trdj2x Hook-title2-dfih78"
          >
            R
            <span
              class="Hook-small-e5sp6t"
            >
              IDERS
            </span>
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body2 Hook-text-t3rl6s Hook-subtitle-1wf8bht"
          >
            hero.subtitle
          </p>
          <div
            class="Hook-containerNewsletter-hfs42l"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth Hook-button-mqnrj"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                newsletter.signUp
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Hook-root-xpbc82"
    >
      <div
        class="Hook-img-2zpuhd gatsby-image-wrapper"
        style="position: absolute; overflow: hidden;"
      >
        <div
          style="width: 100%;"
        />
        <picture>
          <img
            alt="wave"
            sizes="mockString"
            src="mockString"
            srcset="mockString"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; opacity: 0;"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;img sizes="mockString" srcset="mockString" src="mockString" alt="wave" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
      <div
        class="Hook-container-gxkfpg"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom Hook-text-t3rl6s Hook-title-trdj2x"
        >
          E
          <span
            class="Hook-small-e5sp6t"
          >
            ISBACH
          </span>
        </h2>
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom Hook-text-t3rl6s Hook-title-trdj2x Hook-title2-dfih78"
        >
          R
          <span
            class="Hook-small-e5sp6t"
          >
            IDERS
          </span>
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body2 Hook-text-t3rl6s Hook-subtitle-1wf8bht"
        >
          hero.subtitle
        </p>
        <div
          class="Hook-containerNewsletter-hfs42l"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth Hook-button-mqnrj"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              newsletter.signUp
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
